embark.el

```diff
@@ -1975,42 +2001,42 @@ ARG is the prefix argument."
          (orig-type (plist-get transformed :orig-type))
          (dir (embark--default-directory))
          (candidates
-          (cl-mapcar
-           (lambda (cand orig-cand)
-             (list :type type :orig-type orig-type
-                   :target (if (eq type 'file) (expand-file-name cand dir) cand)
-                   :orig-target orig-cand))
-           (plist-get transformed :candidates)
-           (plist-get transformed :orig-candidates)))     
+          (or (cl-mapcar
+               (lambda (cand orig-cand)
+                 (list :type type :orig-type orig-type
+                       ;; TODO The file special casing here seems odd.
```

Comment:
The problem is with `default-directory`: when you are completing file names in the minibuffer you can naviagte to a different directory, when you do that the candidates are just the last component of the path, and the directory you navigated too is not the `default-directory` of the minibuffer or of the target buffer. This causes problems in several places in Embark.

Maybe the best solution would be to switch the candidate collectors for the various UIs to return full paths, and then just relativize them again in embark-collect and embark-export. The target finder for the minibuffer does use full paths so embark-act doesn't have to worry about this, it's just the candidate collectors that use, well, completion candidates instead of making them "full candidates".

Reply:
Makes sense. But what I find confusing is that for `embark-export` you somehow fix the `default-directory` such that it reflects the modified path? Couldn't we use the same `default-directory` fix for `embark-act-all`. Of course the alternative would be to switch to full candidates (and abbreviating them again in dired). Probably this would be my preferred solution but I am not sure about potential edge cases.